Forum postoptions bug

rica-carv opened the pull request:
### Motivation and Context
Prevent duplicated delete buttons when in moderator mode

### Description
Added moderator check

### How Has This Been Tested?
E107 latest github

### Types of Changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist
- [x] My code adheres to the e107 [code style standard](https://github.com/e107inc/e107/wiki/e107-Coding-Standard).
- [x] I have read the [**contributing** document](https://github.com/e107inc/e107/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/e107inc/e107/tree/master/e107_tests) to cover my changes.
- [ ] All new and existing tests pass.

rica-carv committed:
Update view_shortcodes.php

rica-carv committed:
Update view_shortcodes.php

rica-carv committed:
Update view_shortcodes.php

Jimmi08 commented:
@rica-carv  
Where is related issue to this PR?  I mean - there is template for any bug.  Any PR should have issue or discussion at first.  PR without explanation shoudln't be accepted.  

Yes, you are fixing problem for you, but others can have other opinion.  PR should be created only if there is approved approach to solve issue like this. 

Your PR is related to this: 

![image](https://github.com/user-attachments/assets/da8e547d-1852-4d8d-bb4b-ac00ef8ebb05)


If I got it right, you are hidding delete button if user is moderator.  This is not correct for me.

User should always had option to delete post if:
- it is its own post, 
- if it is the last in the thread 
- if post is not the initial post of the topic)

Moderator should see delete button, if:
- it is not their own post (there is user button already)  - 
- rules: only show delete button when post is not the initial post of the topic and not user  

Do you see what I mean?  It is opposite approach then yours, this should be discussed and only after final solution PR should be created.  

To complicate things:
-  what if rules for deleting are different for user and moderators (or will be in future)?  And moderator wants to delete post as user (with user rules only)?  Then both buttons are needed. 

So, yes, this is something what should be solved, but not by direct PR, by discussing and making rules.  For example, I don't know why only last post can be deleted by moderator.  

This is example of next approach (changed label "mod" - to let know you are deleting post as moderator):
![image](https://github.com/user-attachments/assets/7ea9fd86-aafc-4fe5-bd73-177163d039dc)

Just my 2 cents.

rica-carv commented:
> Moderator should see delete button, if:
> 
>     * it is not their own post (there is user button already)  -

Moderator already can see the user buttons, so why had duplicated buttons that do the same thing? To add confusion?

I know that the forum plugin needs a major rework, but it's scheduled to do, but....

Jimmi08 commented:
> > Moderator should see delete button, if:
> > ```
> > * it is not their own post (there is user button already)  -
> > ```
> 
> Moderator already can see the user buttons, so why had duplicated buttons that do the same thing? To add confusion?
> 
> I know that the forum plugin needs a major rework, but it's scheduled to do, but....

You prefer delete button in moderator's section, I prefer delete button in user section and hide it for moderator .  The same thing but different approach.

rica-carv committed:
Merge branch 'e107inc:master' into forum_postoptions-bug

rica-carv committed:
Merge branch 'e107inc:master' into forum_postoptions-bug

rica-carv committed:
Merge branch 'e107inc:master' into forum_postoptions-bug

rica-carv committed:
Merge branch 'e107inc:master' into forum_postoptions-bug

rica-carv committed:
Merge branch 'e107inc:master' into forum_postoptions-bug

rica-carv committed:
Merge branch 'e107inc:master' into forum_postoptions-bug

rica-carv committed:
Merge branch 'e107inc:master' into forum_postoptions-bug

e107help committed:
style(forum): drop the commented-out scaffolding from the delete guard

The moderator check was folded into the outer condition, which left the
inner `if` and its braces behind as comments. Fold the explanatory
comment in with it so the guard reads as one rule.
